src/libbson/src/bson/bson.c

```diff
@@ -3584,7 +3584,19 @@ _bson_iter_validate_document (const bson_iter_t *iter,
       state->phase = BSON_VALIDATE_PHASE_LF_REF_KEY;
    }
 
-   (void) bson_iter_visit_all (&child, &bson_validate_funcs, state);
+   bson_iter_visit_all_v2 (
+      &child, &bson_validate_funcs, BSON_ITER_VISIT_NOFLAGS, state);
+
+   if (child.err_off > 0 && state->err_offset < 0) {
+      // Iteration on a direct element of 'child' failed
+      state->err_offset = child.err_off;
```

Comment:
There is currently no test coverage of this branch.

Reply:
Actually, I believe the addition of `BSON_ITER_VISIT_NOFLAGS` makes this branch unreachable, as `child.err_off` is only assigned when `visit_all` fails to validate an element, and we're now disabling all of those checks. Now I'm getting more suspicious. I'll add more tests, just to make sure.